proxy/http/HttpSM.cc

```diff
@@ -4936,6 +4936,12 @@ HttpSM::do_http_server_open(bool raw)
       call_transact_and_set_next_state(HttpTransact::Forbidden);
       return;
     }
+
+    if (HttpTransact::is_server_negative_cached(&t_state) == true &&
+        t_state.txn_conf->connect_attempts_max_retries_dead_server < 0) {
```

Comment:
Shouldn't that be `<= 0`? Or is 0 special?

Reply:
0 means that it will retry 1 time.  I didn't change the current difficult to understand behavior (see above @maskit comment).